[Charlton Lucian Tan Kian Ren] iP

charltonator opened the pull request:
# DUKE

Duke allows you to keep track of your daily tasks. Why waste memory space in your brain when Duke can do it for you?

As Richard Branson said:

> Remember the to-do list but don't forget the to-be list.

* Text-based
* Simple and straightforward to use
* Just all-round amazing

It's simple! you just need to follow these steps!

1. Download the file from [here!](https://github.com/charltonator/ip/releases/tag/v0.1)
2. Double click the file and open it.
3. add your tasks by following the instructions given.
4. let Duke manage your tasks! 🙃

Best part? Duke is **COMPLETELY FREE**

Features of Duke:

- [x] Manage your todo tasks
- [x] Manage your tasks with deadlines
- [x] Manage your events! 

If you're a java programmer, you might want to use the main method to practice Java!

```java
 public static void main(String[] args) {
        GreetingBot newBot = new GreetingBot();
        newBot.startBot("./data/list.txt");
        }
```

Commit:
Add Gradle support

charltonator committed:
Level-1

charltonator committed:
Level-1

charltonator committed:
Level-1

charltonator committed:
no message

charltonator committed:
Merge branch 'master' of https://github.com/charltonator/ip

charltonator committed:
Merge branch 'master' of https://github.com/charltonator/ip

# Conflicts:
#	src/main/java/Duke.java

charltonator committed:
removed alot of excess lines in duke

charltonator committed:
Level-2

charltonator committed:
Level-3

charltonator committed:
Level-4

charltonator committed:
Level-5

charltonator committed:
Level-6:

charltonator committed:
Level-6

charltonator committed:
Level-6

charltonator committed:
Do Level-7

charltonator committed:
Remove redundant getInfo method in tasks

charltonator committed:
Add Level-8 Dates and Times

charltonator committed:
A-MoreOOP

charltonator committed:
A-MoreOOP Ensure each action is saved to list after execution

charltonator committed:
Ensure each action is saved to list after execution

charltonator committed:
Merge branch 'master' of https://github.com/charltonator/ip

charltonator committed:
Add all classes to package duke under src

charltonator committed:
move duke package back to src/main/java

charltonator committed:
Add JUnit test files for Parser and Todo

charltonator committed:
Add JavaDocs

charltonator committed:
Rename done variable in tasks to isDone

charltonator committed:
Add find function

charltonator committed:
Merge branch 'branch-A-CodingStandard'

charltonator committed:
Merge branch 'branch-Level-9'

charltonator committed:
Create duke package and test folder

charltonator committed:
Update updateData method in Storage

charltonator committed:
Update README.md

charltonator committed:
Add Launcher

charltonator committed:
Merge branch 'master' of https://github.com/charltonator/ip

charltonator committed:
Merge remote-tracking branch 'origin/add-gradle-support'

charltonator committed:
Add Gradle

charltonator committed:
Run checkstyle

charltonator committed:
Add Level-10

charltonator committed:
Add all functionalities into GUI

adamozh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+    private final LocalDateTime date;
+
+    /**
+     * Constructor.
```
I believe there should be a empty line here! According to the Java Coding Standard.

adamozh commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,26 @@
+package duke;
+/**
+ * A class that is an exception to be thrown for errors relating to the Duke.
+ */
+
```
I believe there should be a line after package, and the header comment should be right above the beginning of the class declaration

adamozh commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+        this.date = date;
+    }
+
+    /**
+     * toString method.
+     * @return a String.
+     */
```
I don't think you need to add comments for toString methods!

adamozh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,43 @@
+
+
+
+    }
```
This bracket seems misaligned!

adamozh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    /**
+     * Update data in list file.
```
In general, method header comments should start with singular verb e.g. returns, makes, updates instead of return, make, update

adamozh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,87 @@
+package duke;
+
+
+import java.io.*;
```
I believe you should state the classes explicitly instead of using *, perhaps your IDE was not set up properly?

adamozh reviewed (commented):
LGTM, only some minor formatting things 👍

wengYing227 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+
+    /**
+     * Constructor.
+     * @param title title of the task.
```
Is it better if you capitalize the second _title_?

wengYing227 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Constructor.
+     * @param title title of the task.
+     * @param date time of the task.
```
Similar as above

wengYing227 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * An Event class that inherits from Task and includes a time for the Task.
+ */
+
```
Maybe you can try to delete the space after comment to make it more uniform

wengYing227 commented on src/main/java/duke/GreetingBot.java:
```diff
@@ -0,0 +1,120 @@
+        return this.ui.greet();
+    }
+
+    public String loadList(String filePath) {
```
You can add comment to _startBot_ and _loadList as well_

wengYing227 commented on src/main/java/duke/GreetingBot.java:
```diff
@@ -0,0 +1,120 @@
+     */
+    protected String store(String input) {
+        Parser parser = new Parser();
+            try {
```
Indentation space should be 4.

wengYing227 reviewed (commented):
LGTM. :thumbsup:There are only a few styling problems to fix.

charltonator committed:
A-Gradle

charltonator committed:
Fix checkstyle for completed level-10

charltonator committed:
Merge branch 'branch-Level-10'

charltonator committed:
Add Assertions

Add a few lines to ensure program works as intended with Assertions.
This is to ensure along the way the program does not suddenly break.

charltonator committed:
Refactor repetitive lines

In newDeadline and newEvent methods, lines that parse the date input are
repetitive. To make code cleaner, refactor the repetitive lines into new
method dateParser.

charltonator committed:
Add Undo Function

charltonator committed:
Merge pull request #1 from charltonator/branch-A-Assertions

Add Assertions

charltonator committed:
Merge branch 'master' into branch-A-CodeQuality

charltonator committed:
Merge branch 'master' into branch-C-Undo

charltonator committed:
Merge pull request #2 from charltonator/branch-A-CodeQuality

Merge branch-A-CodeQuality

charltonator committed:
Merge branch 'master' into branch-C-Undo

charltonator committed:
Add Ui screenshot and updated README in docs

charltonator committed:
Set theme jekyll-theme-hacker
